### ISSUE TITLE
Freeipa krbPasswordExpiration

### ISSUE THREAD

**Issue by purisev** [ltb-project/self-service-password#792]
Hello.
Have an installation of utility with combination of Freeipa.
So, when user change password it is always expired.
Expiration of passwords checks with user attribute called krbPasswordExpiration.
It looks like: krbPasswordExpiration: 20230803122541Z 
Can you add please in config analog of shadowExpire for FreeIPA users to change krbPasswordExpiration attribute with a preset value in days ?
Thank you.

**Comment by andyfsimon** [comment]
Same here - we would like to implement this to allow our IPA users to reset their password (and to set it for the first time) but it's kind of silly that they have to come up with two valid passwords each time.

**Comment by andyfsimon** [comment]
For the time being, I resolved by allowing the password reset user ("manager") to not force a password change upon the next login, as described here:
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/managing_idm_users_groups_hosts_and_access_control_rules/managing-user-passwords-in-idm_managing-users-groups-hosts#enabling-password-reset-in-idm-without-prompting-the-user-for-a-password-change-at-the-next-login_managing-user-passwords-in-idm

**Comment by coudot** [comment]
Waiting for an official option in SSP, you can for now use a posthook:
https://self-service-password.readthedocs.io/en/stable/config_preposthook.html

Just create a small script that will update krbPasswordExpiration with current date

For the official feature, you're welcome to propose a contibution, or contact us directly to help financing the development.

**Comment by andyfsimon** [comment]
Indeed! That was my initial attempt and while searching how I could do just that I ran into the Red Hat documentation above. Bear in mind, for IPA it only applies when you use the "manager" account (as in the case of reset password) because the "user" will update just fine and not trigger the expiry.

Thing is, if I understand correctly, krbPasswordExpiration should be updated with the date of the next expiration (so we would likely also need to check what is the policy applying in IPA for that user) - then assuming the host running SSP is able to run ipa commands, it should be doable by using the "manager" account in the script as well.

As for integrating in SSP itself, I am unsure whether we can come up with an "update_ipa_expiry = true" or "ipa_chpw_at_next_login = false" fields, but will see whether I can come up with something, but php ain't my daily driver, unfortunately! Is FreeIPA a common occurrence among your users?

But while we're at it: thank you for your work.

**Comment by mvc-sebastian-schueler** [comment]
I got a fix to that issue and would like to open a pull request to contribute.

**Comment by coudot** [comment]
> I got a fix to that issue and would like to open a pull request to contribute.

Of course, please propose your PR

**Comment by mvc-sebastian-schueler** [comment]
> > I got a fix to that issue and would like to open a pull request to contribute.
> 
> Of course, please propose your PR

Can't do: "Pull request creation failed. Validation failed: must be a collaborator"

**Comment by coudot** [comment]
From which repository are you trying to create PR ?

**Comment by mvc-sebastian-schueler** [comment]
> From which repository are you trying to create PR ?

https://github.com/ltb-project/self-service-password

On master the "Create Pull Request" button is not clickable, so i tried on branch 1.7 as it is the actual version. On the latter i got the error message see above.

**Comment by coudot** [comment]
You need to fork the repository first and propose a PR from your fork
